pkg/controller/clusterautoscaler/clusterautoscaler_test.go

```diff
@@ -125,3 +132,120 @@ func TestAutoscalerArgs(t *testing.T) {
 		}
 	}
 }
+
+type MockReconciler struct {
+	getAutoscalerOk bool
+	gASerrType      error
+	configVersion   string
+	isDCCResult     bool
+	calledIsDCC     bool
+}
+
+func (r *MockReconciler) GetAutoscaler(_ *autoscalingv1alpha1.ClusterAutoscaler) (*appsv1.Deployment, error) {
+	dep := &appsv1.Deployment{}
+
+	if !r.getAutoscalerOk {
+		return nil, r.gASerrType
+	}
+	var m map[string]string
+	m = make(map[string]string)
+	dep.ObjectMeta.Annotations = m
+	dep.ObjectMeta.Annotations["release.openshift.io/version"] = r.configVersion
+	return dep, nil
+}
+func (r *MockReconciler) isDeploymentControllerCurrent(dep *appsv1.Deployment) (bool, error) {
+	r.calledIsDCC = true
+	return r.isDCCResult, nil
+}
+
+// This test ensures we can actually get an autoscaler with fakeclient/client.
+// fakeclient.NewFakeClientWithScheme will os.Exit(1) with invalid scheme.
+func TestCanGetca(t *testing.T) {
+	tscheme := runtime.NewScheme()
+	apis.AddToScheme(tscheme)
+	ca := NewClusterAutoscaler()
+	_ = fakeclient.NewFakeClientWithScheme(tscheme, ca)
```

Comment:
Drop the temporary; `_ = fakeclient.NewFakeClientWithScheme(tscheme, NewClusterAutoscaler())`.